### PR TITLE
Add spatial analog algos from flyingpigeon

### LINKED ISSUE #496: Migrate analog algorithms from flyingpigeon
Flyingpigeon has algorithms for analog computations. We want to migrate them to xclim and use xarray and dask.

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #496 
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [x] `bumpversion (minor / major / patch)` has been called on this branch
- [x] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 
Adapts (copies) the spatial analogues algorithms from flyingpigeon into xclim. For now, I haven't done much more then writing a `spatial_analog` function that wraps the metrics in an `apply_ufunc` call. I tried numba, but all metrics use scipy stuff that numba doesn't understand, so the current solution is to have `vectorize=True` in `apply_ufunc`.

* **Does this PR introduce a breaking change?** 
No

### CODE REVIEW

**Comment by aulemahal** [comment]
This might be missing a notebook? That can also be done later.

File: xclim/analog.py [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+
+
+# ---------------------------------------------------------------------------- #
+# -------------------------- Utility functions ------------------------------- #
+# ---------------------------------------------------------------------------- #
+
+
+def standardize(x, y):
+    """
+    deviation.
```
Comment (huard) [inline xclim/analog.py:163]: Missing description.

**Review by huard — APPROVED** [review]
Nice job. Agree about the notebook, let's create another issue for it, as well as another issue for the missing algorithms (low-priority).

**Comment by aulemahal** [comment]
@huard  I have reunited both decorators in one. I thought that "drop when NaN" logic should be the same for all metrics. I.e. : if any value of either distributions (x, y) is NaN, return NaN, don't even bother to try the computation.  For some metric functions, doing otherwise was failing, for other it was returning np.Inf. I think it is now more consistent.

**Comment by huard** [comment]
Good to go from my PoV. Let's defer further improvements with the notebook drafting. 
Thanks !

**Pull request merged**